examples/mixture/plot_gmm_sin.py

```diff
@@ -78,4 +58,36 @@
     plt.xticks(())
     plt.yticks(())
 
+
+# Number of samples per component
+n_samples = 100
+
+# Generate random sample following a sine curve
+np.random.seed(0)
+X = np.zeros((n_samples, 2))
+step = 4 * np.pi / n_samples
+
+for i in xrange(X.shape[0]):
```

Comment:
It is not the classical Py2 `xrange` but the one from `sklearn.externals.six.moves`.
I've used `range` instead.